### PR TITLE
feat(metrics): New normalization of keys, values, units

### PR DESCRIPTION
**Pull request by sentrivana**
See https://develop.sentry.dev/sdk/metrics/#normalization

Closes https://github.com/getsentry/sentry-python/issues/2937

---

## General Notes

Thank you for contributing to `sentry-python`!

Please add tests to validate your changes, and lint your code using `tox -e linters`.

Running the test suite on your PR might require maintainer approval. Some tests (AWS Lambda) additionally require a maintainer to add a special label to run and will fail if the label is not present.

#### For maintainers

Sensitive test suites require maintainer review to ensure that tests do not compromise our secrets. This review must be repeated after any code revisions.

Before running sensitive test suites, please carefully check the PR. Then, apply the `Trigger: tests using secrets` label. The label will be removed after any code changes to enforce our policy requiring maintainers to review all code revisions before running sensitive tests.

### CODE REVIEW

**Review by szokeasaurusrex — APPROVED** [review]
Looks good, please see my question though

File: sentry_sdk/metrics.py [hunk 1]
```diff
@@ -67,6 +73,25 @@
     ]
 )
 
+_sanitize_unit = partial(re.compile(r"[^\w]+").sub, "")
+_sanitize_metric_key = partial(re.compile(r"[^\w\-.]+").sub, "_")
+_sanitize_tag_key = partial(re.compile(r"[^\w\-.\/]+", re.UNICODE).sub, "")
```
[the remark this answers is not in the document]
Comment (sentrivana) [reply sentry_sdk/metrics.py:70]: That's a good question. The [docs](https://develop.sentry.dev/sdk/metrics/#normalization) seem to imply unit and metric key/name should also be unicode-friendly (maybe @cleptric can confirm if this is actually the case). If that's true we should add it to the other methods too.

[the remark this answers is not in the document]
Comment (sentrivana) [reply sentry_sdk/metrics.py]: Clarified with @cleptric now -- unicode is only supported in tag values and nowhere else, so removed the `re.UNICODE` completely and updated the [regexes to the new version](https://develop.sentry.dev/sdk/metrics/#normalization) which doesn't use `\w` (which would be unicode by default in Python 3).

[the remark this answers is not in the document]
Comment (szokeasaurusrex) [reply sentry_sdk/metrics.py]: @sentrivana Can we add a test that ensures that characters (like á ë ß ő, etc.) are filtered out by these sanitization functions? This way, we can protect against someone trying to change these regexes back to using `\w` without realizing that we intentionally are not using `\w` here (perhaps, a code comment could also help).

[the remark this answers is not in the document]
Comment (sentrivana) [reply sentry_sdk/metrics.py]: Good idea -- we already had one test case for the tag keys, added one for metric names/keys and units, too.

**Review by antonpirker — APPROVED** [review]

**Pull request merged**